### PR TITLE
Loosen dataset path restrictions

### PR DESCRIPTION
**Pull request by craigds**
## Description
The rules for dataset paths added in #501 were a bit too unwieldy and
hard to follow.
Since we didn't have any clear reason for forbidding initial characters
in dataset paths, we discussed and decided to relax the restriction, so
that now the first character of a dataset path has almost the same
restrictions as the second and later characters. (However, starting with
a `/` or `.` is still forbidden)

Additionally this change fixes an unintentional behaviour in `import`
where a `.` in a dataset path was replaced with a `/`. This was intended
to import schema-qualified database names in a hierarchy to avoid name
conflicts when importing with `--all-tables`.

However we felt that was a misfeature; the user probably doesn't want to
import schema-qualified names, and if they do they can do so explicitly.
So this change removes that behaviour also.
## Checklist:

- [x] Have you reviewed your own change?
- [x] Have you included test(s)?
- [x] Have you updated the [changelog](https://github.com/koordinates/kart/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Comment by olsen232** [comment]
The code that causes a fully-qualified[1] table name to sometimes arrive as the default dataset path is still intact - this happens whenever someone tries to import from a sqlalchemy datasource without specifying the the schema-name as part of the datasource, which means they have to specify a schema-name + table-name for each table they wish to import.
The code previously imported such tables to a dataset called `schema-name/table-name`, now it will import them to `schema-name.table-name`.
If you like, you can punt this for a bit - or I can take it, since I put together the sqlalchemy import source in the first place and so it's my fault this bit of it is a bit esoteric.
 
[1] By fully-qualified I mean things like `schema-name.table-name` - or for MySQL, `db-name.table-name`. Whether this is actually fully qualified is up for debate.

**Comment by craigds** [comment]
> The code that causes a fully-qualified[1] table name to sometimes arrive as the default dataset path is still intact

isn't that [this change](https://github.com/koordinates/kart/pull/502/files#diff-30ee65974df04dc5209fa0c5d85cbabb846f636a57bb212608082b7766545397R154) ?

I confess I didn't actually _try it_ ;) but that's what that code seems to do

**Review by olsen232 — APPROVED** [review]
